### PR TITLE
doc: Add reference page for code intelligence envarrs

### PR DESCRIPTION
**Pull request by efritz**

### CODE REVIEW

File: doc/code_intelligence/references/envvars.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+| Name | Default | Description |
+| ---- | ------- | ----------- |
+| `_MAX_AGE_FOR_NON_STALE_BRANCHES` | `2160h` (about 3 months) | The age after which a branch should be considered stale. Code intelligence indexes will be evicted from stale branches. |
+| `_MAX_AGE_FOR_NON_STALE_TAGS` | `8760h` (about 1 year) | The age after which a tagged commit should be considered stale. Code intelligence indexes will be evicted from stale tagged commits. |
+| `_COMMIT_GRAPH_UPDATE_TASK_INTERVAL` | `10s` | The frequency with which to run periodic codeintel commit graph update tasks. |
+
+### `codeintel-auto-indexing`
+
+THe following variables influence the behavior of the [`codeintel-auto-indexing` worker task](../../admin/workers.md#codeintel-auto-indexing).
```
Comment (Strum355) [inline doc/code_intelligence/references/envvars.md:60]:
```suggestion
The following variables influence the behavior of the [`codeintel-auto-indexing` worker task](../../admin/workers.md#codeintel-auto-indexing).
```

line 50 too

File: doc/code_intelligence/references/envvars.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Code intelligence environment variables
+
+All of the following environment variables are prefixed with `PRECISE_CODE_INTEL`.
+
+## frontend
+
+The following are variables are read from the `frontend` service to control code intelligence behavior exposed via the GraphQL API.
```
Comment (Strum355) [inline doc/code_intelligence/references/envvars.md:7]: Bit confusing on the "code intelligence behavior exposed via the GraphQL API" part when most of these seem related to db migrations. Am i missing something here

Comment (efritz) [reply doc/code_intelligence/references/envvars.md:7]: Yeah that's something we're actively trying to migrate

**Review by Strum355 — APPROVED** [review]

**Comment by DaedalusG** [comment]
noice

**Pull request merged**